mne_bids/sidecar_updates.py

```diff
@@ -231,6 +255,17 @@ def update_anat_landmarks(
             f'bids_path. Tried the following filenames: '
             f'{", ".join([p.name for p in tried_paths])}')
 
+    if not isinstance(landmarks, DigMontage):  # it's pathlike
+        if fs_subject is None:
+            fs_subject = bids_path.subject
```

Comment:
don't do such magic and do what the docstring says. You must provided the fs_subject in this case

Reply:
I'm actually **doing** what the docstring says, I just copied this exact behavior from ``get_head_mri_trans()`` for consistency:

https://github.com/mne-tools/mne-bids/blob/b26f062a4e1c68fc83ec69df44cd345841a5aed1/mne_bids/read.py#L919-L920

I don't like it either, but I'd say we either do this in both functions or in neither.

Ok to  remove the magic here and deprecate this behavior in `get_head_mri_trans()?